Classes/DataProcessing/FlexFormProcessor.php

```diff
@@ -99,16 +99,15 @@ public function process(ContentObjectRenderer $cObj, array $contentObjectConfigu
             $fieldName = 'pi_flexform';
         }
 
-        if (!$processedData['data'][$fieldName] && !$processedData[$fieldName]) {
+        // processing the flexform data
+        $originalValue = $processedData['data'][$fieldName] ?? $processedData[$fieldName] ?? null;
+        if ($originalValue === null) {
```

Comment:
you could add checks for empty string and array, so we can early return and !empty later on will be redundant